clang/lib/Driver/ToolChains/Haiku.cpp

```diff
@@ -128,8 +253,14 @@ void Haiku::addLibCxxIncludePaths(const llvm::opt::ArgList &DriverArgs,
                    concat(getDriver().SysRoot, "/boot/system/develop/headers/c++/v1"));
 }
 
-void Haiku::addLibStdCxxIncludePaths(const llvm::opt::ArgList &DriverArgs,
-                                     llvm::opt::ArgStringList &CC1Args) const {
-  addLibStdCXXIncludePaths(concat(getDriver().SysRoot, "/boot/system/develop/headers/c++"),
-                           getTriple().str(), "", DriverArgs, CC1Args);
+void Haiku::addClangTargetOptions(const ArgList &DriverArgs,
+                                    ArgStringList &CC1Args,
+                                    Action::OffloadKind) const {
+  if (!DriverArgs.hasFlag(options::OPT_fuse_init_array,
```

Comment:
Does Haiku need `.ctors`/`/`.dtors` compatibility?

Reply:
Looking closer it appears this is not necessary.